contracts/PodCastNFT.sol

```diff
@@ -260,11 +260,55 @@ contract PodCastNFT is ERC721, Ownable {
             "You cannot answer twice"
         );
         userAttribute[msg.sender].point += 100;
+        string memory podcastRole = checkPodcastRole();
+        if (bytes(podcastRole).length != bytes("").length) {
+            roles[msg.sender].push(podcastRole);
+        }
         userAttribute[msg.sender].claimableToken += 100e18;
         userAttribute[msg.sender].answeredAt = block.timestamp;
         emit CheckedAnswer(msg.sender, userAttribute[msg.sender].answeredAt);
     }
 
+    function checkPodcastRole() private returns (string memory) {
```

Comment:
It is a convention I am promoting on my own, but it might be preferable to have the internal functions under L:91 with underscores.

https://github.com/henkaku-center/podcast-nft/blob/c84766092d97be609c7262aa61544837eda91476/contracts/PodCastNFT.sol#L91